public/app/features/alerting/state/actions.ts

```diff
@@ -17,3 +20,47 @@ export function togglePauseAlertRule(id: number, options: { paused: boolean }):
     dispatch(getAlertRulesAsync({ state: stateFilter.toString() }));
   };
 }
+
+export function createNotificationChannel(data: any): ThunkResult<void> {
+  return async dispatch => {
+    await getBackendSrv()
+      .post(`/api/alert-notifications`, data)
+      .then(() => {
+        appEvents.emit(AppEvents.alertSuccess, ['Notification created']);
+        dispatch(updateLocation({ path: 'alerting/notifications' }));
+      })
+      .catch(error => {
+        appEvents.emit(AppEvents.alertError, [error.data.error]);
+      });
```

Comment:
nit: 

```suggestion
    try {
        await getBackendSrv().post(`/api/alert-notifications`, data);
        appEvents.emit(AppEvents.alertSuccess, ['Notification created']);
        dispatch(updateLocation({ path: 'alerting/notifications' }));
    } catch (error) {
        appEvents.emit(AppEvents.alertError, [error.data.error]);
    }
```

I would probably keep this promised-based or async/await instead of mixing both.